core/src/main/java/org/infinispan/remoting/transport/jgroups/SiteMasterPickerImpl.java

```diff
@@ -18,11 +18,11 @@
 
    @Override
    public Address pickSiteMaster(List<Address> site_masters, Address original_sender) {
-      return site_masters.get(0);
+      return (site_masters == null || site_masters.isEmpty()) ? null : site_masters.get(0);
```

Comment:
I haven't been able to check, but are we sure this case won't cause a NPE somewhere else now? Since this method never returned a non null value before.

Reply:
I fixed this in jgroups to skip invoking this method if the list is empty. 
If I recall correctly, returning null if fine